explorer-ui/src/components/ProfileView/ProfileDisplay/ProfileDisplay.jsx

```diff
@@ -28,59 +28,154 @@ export default function ProfileDisplay({userProfile}) {
+  //     getCurrLocation(userProfile)
+  //     // getUserCoords(profiles[0])
+  // }, [profiles, userProfile])
   useEffect(() => {
-      getCoordinates(profiles);
-      // getUserCoords(profiles[0])
-  }, [])
-  const getCoordinates = (profiles) => {
-      if (profiles) {
-        for (let i = 0; i < profiles.length; i++) {
-          if(( profiles[i].address != null || profiles[i].address!="")){ //&& profiles[i].preferenceInfo.visibility =="yes"){
-          Geocode.fromAddress(profiles[i].address).then(
+    console.log("second rendering");
+    getMatches(profiles); 
+    setMatchesLoadingComplete(true)
+    getCurrLocation(userProfile)
+  }, [profiles, userProfile, matchesLoadingComplete, matchesSortingComplete, coordsLoaded,])
+  useEffect(() =>{
+    rankMatches(matches)
+    setMatchesSortingComplete(true)
+    getCoordinates(rankedMatches)
+  }, [matches])
+    // TODO: Create a function that filters through the list of users to get the matches of this specific user
+    const getMatches = () => {
+      console.log("getting matches")
+      const profilesToAdd = new Set();
+      for (let i = 0; i <Object.keys(profiles).length; i++) {
+        console.log("user country: ", userProfile.country),
+          console.log("profile country: ", profiles[i].country)
+        if (
+          profiles[i].country === userProfile.country 
+          &&
+          profiles[i].travelMonth == userProfile.travelMonth &&
+          profiles[i].accomodation == userProfile.accomodation
+        ) {
+          console.log("it's a match!")
+          console.log("profiles[i]: ", profiles[i])
+          // profilesToAdd.push(profiles[i])
+          profilesToAdd.add(profiles[i])
+        }
+      }
+      setMatches([...matches, ...profilesToAdd])
+      console.log("profilesToAdd: ", profilesToAdd)
+      
+    };
+  
+      // Calculate the intersections between interests and attractions
+    function findIntersections(match){
+      console.log("find intersections")
+      if(match.preferenceInfo.interests && userProfile.preferenceInfo.interests){
+        console.log("match.preferenceInfo.interest: ", match.preferenceInfo.interests)
+        let interstIntersection = userProfile.preferenceInfo.interests.filter(x => match.preferenceInfo.interests.includes(x));
+        console.log("userProfile.preferenceInfo.interest: ", userProfile.preferenceInfo.interests)
+        let attractionIntersection = userProfile.preferenceInfo.attractions.filter(y =>  match.preferenceInfo.attractions.includes(y));
+        return {"interestIntersection": interstIntersection, "attractionIntersection": attractionIntersection, "match": match};
+      }
+      console.log("intersection: ", interstIntersection)
+      return {"interestIntersection": {}, "attractionIntersection": {}, "match": match};
+    }
+    
+    const rankMatches = (matches) =>{
+      console.log("rank matches")
+      console.log("matches are: ", matches)
+      if(matches){
```

Comment:
more explicit check -> not null and not empty

Reply:
fixed